docs/writing-tests/testharness.md

```diff
@@ -198,15 +182,84 @@ Use `// META: script=link/to/resource.js` at the beginning of the resource. For
 
 can be used to include both the global and a local `utils.js` in a test.
 
-### Specifying a timeout of long in auto-generated boilerplate tests
+### Specifying a timeout of long
 
 Use `// META: timeout=long` at the beginning of the resource.
 
-### Specifying test [variants](#variants) in auto-generated boilerplate tests
+### Specifying test [variants](#variants)
 
 Use `// META: variant=url-suffix` at the beginning of the resource. For example,
 
 ```
 // META: variant=
 // META: variant=?wss
 ```
+
+## Variants
+
+A test file can have multiple variants by including `meta` elements,
+for example:
+
+```html
+<meta name="variant" content="">
+<meta name="variant" content="?wss">
+```
+
+Test runners will execute the test for each variant specified, appending the corresponding content
+attribute value to the URL of the test as they do so.
+
+There are two utility scripts in that work well together with variants,
+`/common/subset-tests.js` and `/common/subset-tests-by-key.js`, where
+a test that would otherwise have too many tests to complete inside the timeout can be
+split up in ranges of subtests. For example:
```

Comment:
```suggestion
`/common/subset-tests.js` and `/common/subset-tests-by-key.js` are two utility
scripts that work well together with variants, allowing a test to be split up into
subtests in cases when there are otherwise too many tests to complete inside the
timeout. For example:
```